django/db/backends/postgresql/base.py

```diff
@@ -258,12 +258,11 @@ def _nodb_connection(self):
                 "and will use the default database instead.",
                 RuntimeWarning
             )
-            settings_dict = self.settings_dict.copy()
-            settings_dict['NAME'] = settings.DATABASES[DEFAULT_DB_ALIAS]['NAME']
             nodb_connection = self.__class__(
-                self.settings_dict.copy(),
+                {**self.settings_dict, 'NAME': settings.DATABASES[DEFAULT_DB_ALIAS]['NAME']},
```

Comment:
Possibly more readable / common pattern to do `dict(self.settings_dict, NAME= settings.DATABASES[DEFAULT_DB_ALIAS]['NAME'])` ?

Reply:
@akulakov - that may seem to be a more common pattern, but it is only because unpacking generalisations have only been available since Python 3.5, which is now the minimum version supported by Django. Unpacking generalisations also have the advantage that they have a dedicated operation that has less overhead than the function call, `dict()` in this case. The proposed change is more concise and perfectly clear according to the syntax of the language.

Reply:
@pope1ni Thanks! I have nothing to add :+1: .

Reply:
Ok, makes sense!